### PR TITLE
Fix remove liquidity, start decimal refactor for some important math

### PR DESCRIPTION
**Pull request by Brendan-Lucas**
# Description
Fix remove liquidity primarily through correcting the contract address. 
Move some removeLiquidity important math to be handled by Decimals.js instead of standard javascript numbers.

## Sanity Check
Removing a liquidity position should remove the position completely and throw no errors.

### CODE REVIEW

File: src/hooks/useGetUpToDatePositionLiquidityParameters.ts [hunk 1]
```diff
@@ -3,8 +3,11 @@ import { BigNumber } from 'ethers'
 import { useAccount } from 'wagmi'
 import { useOrderBookContract } from 'hooks/useOrderBookContract'
 import { selectActivePosition } from 'recoil/selectors/LPPositions/selectActivePosition'
+import { IUpToDateLiquidityParams } from 'utilities/getUpToDatePositionLiquidity'
 
-export const useGetUpToDatePositionLiquidityParameters = () => {
+export const useGetUpToDatePositionLiquidityParameters = ():
+    | IUpToDateLiquidityParams
```
Comment (nickwang14) [inline src/hooks/useGetUpToDatePositionLiquidityParameters.ts:9]: These params should be moved to a `d.types` file.

I made a comment here: https://github.com/Component-fi/element-fixed-borrow/pull/30/files#r1136493206 as well, to share a stance and opinion about proptype declarations. Would like your opinion on it, maybe we can move in the same direction in both projects

File: src/recoil/selectors/LPPositions/selectCurrentPositionsInformation.ts [hunk 2]
```diff
@@ -38,7 +39,7 @@ export const selectCurrentPositionsInformation =
                         position.upperTick,
                     ) as number,
                     currentPrice: position.currentPrice,
-                    liquidity: position.liquidity,
+                    liquidity: 0,
```
Comment (nickwang14) [inline src/recoil/selectors/LPPositions/selectCurrentPositionsInformation.ts:42]: Is this expected?

File: src/recoil/types.d.ts [hunk 3]
```diff
@@ -96,7 +96,7 @@ type PositionsDataType = {
     id: string
     netPositionGain?: number
     presentLiquidity?: number
-    liquidity: number
+    liquidity: Decimal
```
Comment (nickwang14) [inline src/recoil/types.d.ts:99]: I think we should stick to Number or Decimal and figure out where we want the data structure to flow.

Personally I think selectors should be converting and handling decimal across the board and number conversions should be normalized at Component or render-time with a utility.

I think Blueprint actually has a term/pattern for this, maybe we could borrow from them and be consistent?

File: src/utilities/getUpToDatePositionLiquidity.ts [hunk 4]
```diff
@@ -1,13 +1,14 @@
...
+import { OrderBook } from 'types/ethers-contracts/OrderBook'
+import { reduceDecimalBy10Pow18 } from 'utilities/reduceDecimalBy10Pow18'
 
 export interface IUpToDateLiquidityParams {
     address: string
     baseTokenAddress: string
     bigNumberLowerTick: BigNumber
     bigNumberUpperTick: BigNumber
-    orderBookContract: any
+    orderBookContract: OrderBook
```
Comment (nickwang14) [inline src/utilities/getUpToDatePositionLiquidity.ts:11]: Nice

File: src/utilities/liquidityMathUtil.ts [hunk 5]
```diff
@@ -24,6 +25,23 @@ export const deltaSqrtP = (Pu: number, Pl: number, Pc: number) => {
     ]
 }
 
+export const deltaSqrtPDecimal = (
```
Comment (nickwang14) [inline src/utilities/liquidityMathUtil.ts:28]: I think this is pretty valid and likely should be adopted more consistently (see comment above)

File: src/utilities/reduceDecimalBy10Pow18.ts [hunk 6]
```diff
@@ -2,6 +2,16 @@ import { Decimal } from 'decimal.js'
 import { BigNumber } from 'ethers'
 import { WAD } from 'constants/constants'
 
+export const toDecimal = (
```
Comment (nickwang14) [inline src/utilities/reduceDecimalBy10Pow18.ts:5]: This function should be factored out to another format/normalize data util where we store a list of data handlers for each method. 

Bignumber-> Number/Decimal/String
Decimal -> BigNumber/Number/String
Number -> ...

Maybe we can make an index called NumberConversions or something so the imports and utilities can all be in the same spot.

Good TD Item

File: src/components/Modals/EditLiquidityModal/Views/RemoveLiquidity.tsx [hunk 7]
```diff
@@ -72,89 +71,119 @@ export const RemoveLiquidity = ({
...
-            updatedPositionLiquidityParameters as IUpToDateLiquidityParams,
-        )) as number
-
-        const [baseTokens, quoteTokens] =
-            getAmountOfTokensXAndYWithKnownLiquidity(
-                upperPrice,
-                lowerPrice,
-                lastTradePriceUsd,
-                liquidity,
+        if (updatedPositionLiquidityParameters !== undefined) {
```
Comment (nickwang14) [inline src/components/Modals/EditLiquidityModal/Views/RemoveLiquidity.tsx:74]: Isn't there a timing condition issue here now that you're removing the await?

File: src/components/Modals/EditLiquidityModal/Views/RemoveLiquidity.tsx [hunk 8]
```diff
@@ -72,89 +71,119 @@ export const RemoveLiquidity = ({
...
-                    rel="noreferrer"
-                >
-                    here
-                </a>
-                .
-            </p>
-        )
+            const minBaseBigNumber = BigNumber.from(
+                minBase.floor().toHex(),
+            ).toString()
```
Comment (nickwang14) [inline src/components/Modals/EditLiquidityModal/Views/RemoveLiquidity.tsx:111]: Why do we toString this if it's bignumber? Also how come we're no longer using 
`convertBigNumber`? Like `convertBigNumber( minBase.floor().toHex() )`.toString()

Comment (nickwang14) [reply src/components/Modals/EditLiquidityModal/Views/RemoveLiquidity.tsx:111]: I definitely feel like `bigNumber.from().toString()` could be made into a function, even if locally

Comment (Brendan-Lucas) [reply src/components/Modals/EditLiquidityModal/Views/RemoveLiquidity.tsx:111]: that makes sense

File: src/recoil/types.d.ts (hunk 3 above)
Comment (nickwang14) [reply src/recoil/types.d.ts:99]: Yeah agreed, just want to capture this in the PR, let's open a zenhub ticket

**Review by nickwang14 — APPROVED** [review]
Nicely done

**Pull request merged**